tests/llama_index/test_llama_index_model_export.py

```diff
@@ -471,3 +476,70 @@ def test_save_engine_with_engine_type_issues_warning(model_path):
 
     assert mock_logger.warning.call_count == 1
     assert "The `engine_type` argument" in mock_logger.warning.call_args[0][0]
+
+
+@pytest.mark.skipif(
+    Version(llama_index.core.__version__) < Version("0.11.0"),
+    reason="Workflow was introduced in 0.11.0",
+)
+@pytest.mark.asyncio
+async def test_save_load_workflow_as_code():
+    from llama_index.core.workflow import Workflow
+
+    index_code_path = "tests/llama_index/sample_code/simple_workflow.py"
+    with mlflow.start_run():
+        model_info = mlflow.llama_index.log_model(
+            llama_index_model=index_code_path,
+            artifact_path="model",
+            input_example={"topic": "pirates"},
+        )
+
+    # Signature
+    assert model_info.signature.inputs == Schema([ColSpec(type=DataType.string, name="topic")])
+    assert model_info.signature.outputs == Schema([ColSpec(DataType.string)])
+
+    # Native inference
+    loaded_workflow = mlflow.llama_index.load_model(model_info.model_uri)
+    assert isinstance(loaded_workflow, Workflow)
+    result = await loaded_workflow.run(topic="pirates")
```

Comment:
Not a nit: Let's just make sure to introduce a warning in the docs about Workflows that mention the native use deviation here for an async call :)

Reply:
Absolutely, will add that warning in the documentation PR.